### PR TITLE
Create a PR for PM Tien Tran to review

### PR DESCRIPTION
**Pull request by ttuan-evizi**

### CODE REVIEW

File: src_Car/SuperCar.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Vehicle } from "./Vehicle";
```
Comment (kahnext) [inline src_Car/SuperCar.ts]: do not create bunch of `src_` folder, let's organize it with sub folder instead

File: src_Car/SuperCar.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public setTopSpeed(newTopSpeed: number): void {
+        if (newTopSpeed > 0) {
+            this.topSpeed = newTopSpeed; // Update top speed
+        } else {
+            console.error("Top speed must be positive."); // Log error for invalid input
+        }
+    }
+
+    // Public method to get top speed
+    public getTopSpeed(): number {
```
Comment (kahnext) [inline src_Car/SuperCar.ts]: don't see any override and overload in this inheritance Vehicle <> SuperCar

File: src_Geometry/Triangle.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Shape } from './Shape';
+
+export class Triangle extends Shape {
```
Comment (kahnext) [inline src_Geometry/Triangle.ts]: still don't see any overloading support within this Geometry implementation

File: src_Car/SuperCar.ts (hunk 1 above)
Comment (ttuan-evizi) [reply src_Car/SuperCar.ts]: yes, ill combine them into 1 folder src, with subfolders such as Car, and Geometry

File: src_Car/SuperCar.ts (hunk 2 above)
Comment (ttuan-evizi) [reply src_Car/SuperCar.ts]: added overloading to SuperCar

File: src_Geometry/Triangle.ts (hunk 3 above)
Comment (ttuan-evizi) [reply src_Geometry/Triangle.ts]: added overloading to Triangle, Circle, and other shapes as well

File: src/Geometry/Rectangle.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Shape } from "./Shape";
+
+export class Rectangle extends Shape {
+    constructor(private width: number, private height: number) {
+        super();
+    }    
+    // Overloading
+    calculateArea(): number;
```
Comment (kahnext) [inline src/Geometry/Rectangle.ts]: it's not overload anything. I don't see several methods with same name here.

File: src/Geometry/Triangle.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Shape } from './Shape';
+
+export class Triangle extends Shape {
+    constructor(private base: number, private height: number, private sideA: number, private sideB: number) {
+        super();
+    }
+
+    // Overloading
+    calculateArea(): number;
+    calculateArea(base: number, height: number): number;
```
Comment (kahnext) [inline src/Geometry/Triangle.ts]: where is implementation here? It's just a declaration.

File: src/Geometry/main.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { Circle } from './Circle';
+import { Rectangle } from './Rectangle';
+import { Square } from './Square';
+import { Triangle } from './Triangle';
+import { Trapezoid } from './Trapezoid';
+function main() {
```
Comment (kahnext) [inline src/Geometry/main.ts]: do not spam bunch of main file. How if we just create relevant method with specific name like `geo_test()`, `car_test()`, etc..
Then having only one main file includes above functions.

File: src/Phone/iPhone.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    private storage: number;
+    constructor(os: string, brand: string, model: string, price: number, serial: number, version: number, storage: number) {
+        super(os, brand, model, price, serial,);
+        this.version = version;
+        this.storage = storage;
+    }
+
+    public call(): string;
+    public call(phoneNumber: string): string;
+    public call(phoneNumber?: string) {
```
Comment (kahnext) [inline src/Phone/iPhone.ts:14]: good overloading support

File: src/Phone/SamsungS24.ts
Comment (kahnext) [inline src/Phone/SamsungS24.ts:1]: why do we push empty file.

File: src/Phone/iPhone16.ts
Comment (kahnext) [inline src/Phone/iPhone16.ts:1]: why do we push empty file

File: src/Phone/main.ts
Comment (kahnext) [inline src/Phone/main.ts:1]: it's empty file, please remove.

File: src/Phone/SamsungS24.ts
Comment (ttuan-evizi) [reply src/Phone/SamsungS24.ts:1]: They have been removed. I should have checked before pushing them. My bad.

File: src/Phone/main.ts
Comment (ttuan-evizi) [reply src/Phone/main.ts:1]: They have been removed. I should have checked before pushing them. My bad.

File: src/Phone/iPhone16.ts
Comment (ttuan-evizi) [reply src/Phone/iPhone16.ts:1]: They have been removed. I should have checked before pushing them. My bad.

File: src/Geometry/main.ts (hunk 6 above)
Comment (ttuan-evizi) [reply src/Geometry/main.ts]: I have moved main to root, and created relevant method with specific name like geo_test(), car_test() that will be run using main.

File: src/Geometry/Triangle.ts (hunk 5 above)
Comment (ttuan-evizi) [reply src/Geometry/Triangle.ts]: I have added implementations to these functions.

File: src/Geometry/Rectangle.ts (hunk 4 above)
Comment (ttuan-evizi) [reply src/Geometry/Rectangle.ts]: I have added implementations to these functions.

**Review by kahnext — APPROVED** [review]
LGTM

**Pull request merged**